#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_g++4.1-fixes.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Aloow compilation against g++ 4.1

@DPATCH@

--- octave2.1-2.1.73.orig/liboctave/ArrayN.h
+++ octave2.1-2.1.73/liboctave/ArrayN.h
@@ -63,7 +63,7 @@
   ArrayN (const dim_vector& dv) : Array<T> (dv) { }
 
   ArrayN (const dim_vector& dv, const T& val)
-    : Array<T> (dv) { fill (val); }
+    : Array<T> (dv) { Array<T>::fill (val); }
 
   template <class U>
   explicit ArrayN (const Array2<U>& a) : Array<T> (a, a.dims ()) { }
--- octave2.1-2.1.73.orig/liboctave/DiagArray2.h
+++ octave2.1-2.1.73/liboctave/DiagArray2.h
@@ -125,7 +125,7 @@
     {
       this->dimensions = dim_vector (r, c);
 
-      fill (val);
+      Array<T>::fill (val);
     }
 
   DiagArray2 (const Array<T>& a) : Array<T> (a)
